packages/plugin-ext/src/main/browser/tasks-main.ts

```diff
@@ -196,10 +221,25 @@ export class TasksMainImpl implements TasksMain, Disposable {
             }
         }
 
+        this.convertTaskPresentation(taskDto, presentation);
+
         return Object.assign(taskDto, {
             source: taskDto._source,
             scope: taskDto._scope
         });
     }
 
+    private convertTaskPresentation(task: TaskConfiguration | TaskDto,
+        presentationFrom: TaskOutputPresentation | TaskPresentationOptionsDTO | undefined): void {
+        if (presentationFrom) {
+            const { reveal, panel, ...presentationTo } = presentationFrom;
+            if (reveal) {
+                Object.assign(presentationTo, { reveal: revealKindMap.get(reveal) });
+            }
+            if (panel) {
+                Object.assign(presentationTo, { panel: panelKindMap.get(panel) });
+            }
+            Object.assign(task, { presentation: presentationTo });
+        }
+    }
```

Comment:
I am not particularly fond of that logic... I've made something I would be more comfortable with, please tell me if you prefer it too: https://github.com/alvsan09/theia/compare/task_presentation_plugins...eclipse-theia:mp/task_presentation_plugins-2

Reply:
Hi @marechal-p, 
I like your proposal and have squashed your changes on top as a new commit, 
Thanks !!
/Alvaro